spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,503 @@
+
+```go
+type VirtualMachineScaleSetSpec struct {
+    // +kubebuilder:validation:Required
+    ResourceGroupRef *azcorev1.KnownTypeReference `json:"resourceGroupRef" group:"microsoft.resources.infra.azure.com" kind:"ResourceGroup"`
+    ...
+}
+```
+
+The other pattern is where the owning resource has links to the dependent resources it is expecting to have:
+
+```go
+type RouteTableSpecProperties struct {
+    DisableBGPRoutePropagation bool                          `json:"disableBgpRoutePropagation,omitempty"`
+    RouteRefs                  []azcorev1.KnownTypeReference `json:"routeRefs,omitempty" group:"microsoft.network.infra.azure.com" kind:"Route" owned:"true"`
+}
+```
+
+If the dependent resources aren't there, the status of the owning resource reflects an error explaining that.
+
+## Proposal
+
+### How to represent references
+We construct a new type which is our reference holder. I am calling this approach `AugmentedKnownTypeReference` (real type name TBD), because
+it's similar to the current approach that `k8s-infra` is taking but with a few changes.
+
+```go
+type AugmentedKnownTypeReference struct {
+    Kubernetes KubernetesReference
+    Azure      AzureReference
+}
+
+type KubernetesReference struct {
+    // The "Group" of the owning resources GVK. There may be an annotation on the property 
+    // that specifies Group, in which case this is optional (if supplied it must match the group
+    // on the annotation). For resources which do not have a group annotation, this is required.
+    // TODO: It's possibly going to be confusing to customers when they have to specify this and when they don't...
+    Group string
+
+    // The "Kind" of the owning resources GVK. There may be an annotation on the property 
+    // that specifies Kind, in which case this is optional (if supplied it must match the kind
+    // on the annotation). For resources which do not have a kind annotation, this is required.
+    // TODO: It's possibly going to be confusing to customers when they have to specify this and when they don't... 
+    Kind string
+    
+    // Note that no Version is required here because the objective is to link to a single Kubernetes
+    // object, and in Kubernetes versions are all just views on the same entity. The assumption (which I believe Azure honors
+    // is that every version of the Foo type has the same ID shape (it's fundamentally not the same kind of resource
+    // if the ID has changed). Since all we need to do is look that entity up in etcd and get its name details so that
+    // we can build a fully qualified name, we don't actually need version here.
+
+    // This is the name of the kubernetes resource to reference. 
+    // References across namespaces are not supported.
+    Name string
+
+    // Note that ownership across namespaces in Kubernetes is not allowed, but technically resource
+    // references are. There are RBAC considerations here though so probably easier to just start by 
```

Comment:
I had to google a definition for this - perhaps use the full term, or at least add to a glossary.

Reply:
Added a link at the top of the documented for related reading